subprojects/gradle/module-dependencies-graph/src/main/kotlin/com/avito/module/internal/dependencies/DependenciesGraphBuilder.kt

```diff
@@ -0,0 +1,85 @@
+package com.avito.module.internal.dependencies
+
+import com.avito.logger.LoggerFactory
+import com.avito.logger.create
+import com.avito.module.configurations.ConfigurationType
+import com.avito.module.internal.configurations.ConfigurationCoordinate
+import org.gradle.api.Project
+import org.gradle.api.internal.artifacts.dependencies.DefaultProjectDependency
+
+internal class DependenciesGraphBuilder(
+    private val root: Project,
+    loggerFactory: LoggerFactory
+) {
+
+    init {
+        require(root == root.rootProject)
+    }
+
+    private val logger = loggerFactory.create<DependenciesGraphBuilder>()
+    private val rootProjects = mutableListOf<Project>()
+    private val visited = mutableMapOf<Project, ModuleProjectDependenciesNode>()
+
+    fun buildDependenciesGraph(): Set<ModuleProjectDependenciesNode> {
```

Comment:
I think we don't need `Set<root>` and Gradle's original model with one root is enough.
We should separate concerns. The project's graph has one root, this is what we have in a project.
If a client is interested only in Android application, it can filter them from graph.

Reply:
I think I should rename that class to `AndroidAppsDependenciesGraphBuilder` because now it's implicit behavior

Reply:
Anyway, we can split these responsibilities. I see this graph as probable input for impact analysis.

Reply:
Discussed. It can be done later for further improvements.